lib/middleware/route-handlers.js

```diff
@@ -65,7 +107,9 @@ var setup = function(sourceFiles, cb) {
 
 };
 
-module.exports = function(sourceFiles, cb) {
+module.exports = function(options, cb) {
+    var sourceFiles = options.sourceFiles;
+    autoOptions = options.autoOptions;
```

Comment:
I am not sure what is proper way to do this. Imho blueprint parser should be pulled out of route handler because its a different concern.

This way routes are registered during parsing process. I feel that the list of routes/actions should be a result of parsing and we should register them after.

This is why autoOptions flag is in global scope so I don't mess function parameters.

I will refactor it and let you review it :-)